### PR TITLE
TypeScript Definitions

### PR DESCRIPTION
**Pull request by janhartmann**
Thanks for this great library!

I just started using it and figured out that there was not typings for TypeScript, so I went ahead made them for you. 😀

### CODE REVIEW

**Comment by vladimirvorobiev** [comment]
nice !

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  }
+
+  export interface IDataTableConditionalCellStyles<T> {
+    when: (row: T) => boolean;
+    style: CSSProperties;
+  }
+
+  export interface IDataTablePaginationOptions {
+    rowsPerPageText?: string;
+    rangeSeparatorText?: string;
```
Comment (jbetancur) [inline index.d.ts]: missing `noRowsPerPage` boolean

**Comment by jbetancur** [comment]
LOVE IT and thank you! Just one comment to address in IDataTablePaginationOptions

File: package.json [hunk 2]
```diff
@@ -6,7 +6,8 @@
   "module": "dist/index.es.js",
   "browser": "dist/index.cjs.js",
   "files": [
-    "dist"
+    "dist",
+    "index.d.ts"
```
Comment (jbetancur) [inline package.json]: I'm still getting familiar with ts. Should this just go in dist? Do we need to add a "types" section in package.json?

Comment (janhartmann) [reply package.json]: I've added in `typings`, thanks for the highlight.

File: index.d.ts (hunk 1 above)
Comment (janhartmann) [reply index.d.ts]: Fixed and added more missing properties.

**Comment by jbetancur** [comment]
Thank you! I'm going to cut this into a 4.0 release later today.

**Pull request merged**